pyccel/codegen/printing/cwrappercode.py

```diff
@@ -1040,21 +1128,28 @@ def _print_FunctionDef(self, expr):
         return CCodePrinter._print_FunctionDef(self, wrapper_func)
 
     def _print_Module(self, expr):
-        funcs_to_wrap = expr.funcs
+        funcs_to_wrap = [f for f in expr.funcs if f not in (expr.init_func, expr.free_func)]
```

Comment:
Are `init_func` and `free_func` also contained in the `funcs` list? I thought this was not the case...

Reply:
Yes, the `funcs` are defined from the namespace where `init_func` and `free_func` exist (see [here](https://github.com/pyccel/pyccel/blob/master/pyccel/parser/semantic.py#L1701-L1707))

Reply:
But is that what we really want? Shouldn't we keep `init_func` and `free_func` separate from the other functions, given that they are special? Or do you see the need to include them in the same `funcs` list?

Reply:
In other words, do we really need these two lines?
- [`self.insert_function(init_func)`](https://github.com/pyccel/pyccel/blob/b632a0dd4b648e6f30059fac4a80374fd2c4b997/pyccel/parser/semantic.py#L1676)
- [`self.insert_function(free_func)`](https://github.com/pyccel/pyccel/blob/b632a0dd4b648e6f30059fac4a80374fd2c4b997/pyccel/parser/semantic.py#L1699)

Reply:
I'm not sure. In most cases they need treating in the same way as the other functions
- They still need to be wrapped in the bind c file
- They still need to be printed in the same way in the original file
- Their header still needs to be printed
The only place where they need to be treated differently is in the C API interface where we don't want to expose the call to these 2 functions to python.

My impression is that both implementations would work with a similar number of exceptions in the code, however I suspect that the treatment of `init_func` and `free_func` is more likely to be forgotten. If it is included by default then it can be removed when you see the generated code, if not then there is no reminder

Reply:
I see your point, and I agree that the current solution is probably the safest. 

Can you please add some comment lines in the code as a reminder to the developer? Comments are useful when we insert those functions into `funcs` in the semantic parser, and when we later ignore them in the C wrapper printer.